[OPAL-941] add virtual packages option

RR-USAF opened the pull request:
Resolves OPAL-941

Changes proposed in this pull request:
- adds virtual_package_spec options
- fixes naming bug in download

RR-USAF committed:
feature: added option to specify conda virutal packages

RR-USAF committed:
fix: download now renames filenames to name in manifest

RR-USAF committed:
fix: bumped patch version

RR-USAF committed:
fix: bumped minor version

NM-USAF commented on ib_manifest_util/download_conda_packages.py:
```diff
@@ -36,45 +34,66 @@ def download_packages(
+        except Exception as e:
+            print(f"error loading manifest: {manifest_path.name}")
+            print(str(e))
+            sys.exit(1)
```
Test at `test_download_conda_packages.test_download_package_no_urls_no_manifest` expects FileNotFoundError, not system exit

RR-USAF replied:
I like the new behavior -- fixed the test.

NM-USAF commented on ib_manifest_util/download_conda_packages.py:
```diff
@@ -36,45 +34,66 @@ def download_packages(
+            error_count += 1
+
+    if error_count:
+        sys.exit(1)
```
Test at `test_download_conda_packages.test_download_package_incorrect_url_list` expects this function to exit and the file to not exist. `sys.exit(1)` here breaks that test.

RR-USAF replied:
fixed the tests.

NM-USAF reviewed (changes requested):
Some tests fail
Not covered inline:
- test_download_conda_packages
- test_verify_local_channel_environment

RR-USAF committed:
fix: fixed failing unit tests

cleanup: removed dead code and dead tests
fix: fixes for updates to download_packages

NM-USAF reviewed (approved):
Tests pass now

IM-USAF commented on tests/test_util.py:
```diff
@@ -49,60 +47,6 @@ def test_write_templatized_file_dockefile_default(
         template_dir=template_dir,
     )
 
-
```
Is all of this removed code related to the tests that tested unused code?

RR-USAF replied:
yeah

IM-USAF commented on ib_manifest_util/cli.py:
```diff
@@ -42,13 +42,18 @@ def main() -> None:
     "--output_dockerfile_path",
     help="(Optional) Path to location in which the Dockerfile will be placed",
 )
+@click.option(
```
It's not clear why I would need a virtual package spec. I know you explained this and maybe it would be apparent by the time I'm the person using this tool to update an IB repo. How do I know the schema of the yaml to create? Is there a function to export the default package spec? Would it be useful if there was a suboption that output the default spec in the current user environment?

RR-USAF replied:
this all exists in conda-vendor.  this is essentially a pass-through option.  i don't really think we need it for ib_manifest_util.
